projects/core/src/lib/users/user.service.ts

```diff
@@ -325,6 +330,20 @@ export class UserService {
                 this.sessionQuota$.next(quota);
             });
     }
+
+    private checkOidcParametersAndLogin(): boolean {
+        const params = new URLSearchParams(window.location.search);
+        const sessionState = params.get('session_state');
+        const code = params.get('code');
+        const state = params.get('state');
+
+        if (sessionState && code && state) {
+            this.oidcLogin({sessionState, code, state}).pipe(first()).subscribe();
+            return true;
```

Comment:
This looks… wrong.

Shouldn't you return an `Observable<boolean>` instead?

Reply:
no, this works just like the login worked before. The regular login works asynchronously as weill.